src/compiler/interface.jl

```diff
@@ -115,12 +115,32 @@ as a named tuple.
 julia> y, ∇ = withgradient(/, 1, 2)
 (val = 0.5, grad = (0.5, -0.25))
 
-julia> ∇ == gradient(/, 1, 2)  # explicit mode
+julia> ∇ == gradient(/, 1, 2)
 true
+```
+
+If `f` returns a Tuple or NamedTuple, then it calculates
+`gradient(first∘f, args...)` but returns the whole `f(args...)`:
```

Comment:
maybe explain why we allow this, like "Sometimes it is convenient to return some auxiliary outputs"